java/fory-core/src/main/java/org/apache/fory/Fory.java

```diff
@@ -242,6 +279,8 @@ public void registerSerializer(Class<?> type, Serializer<?> serializer) {
 
   @Override
   public void registerSerializer(Class<?> type, Function<Fory, Serializer<?>> serializerCreator) {
+    Serializer<?> serializer = serializerCreator.apply(this);
+    validateCollectionSerializer(type, serializer);
     if (language == Language.JAVA) {
       classResolver.registerSerializer(type, serializerCreator.apply(this));
```

Comment:
Please validate serializer in ClassResolver, users may use ClassResolver#registerSerializer directly